### ISSUE TITLE
[BUG] Roles not being applied

### ISSUE THREAD

**Issue by eirisdg** [tumbl3w33d/nexus-oauth2-proxy-plugin#26]
**Expected**
I've seted up the authentication with the plugin and Authentik as Oauth Proxy.
Everything is fine (except logout button, but I've seen a PR about it), but role assignment is not working propertly.
I've logged in via Oauth, given 'nx-admin' role, but... it can't see anything.

![image](https://github.com/user-attachments/assets/c7f6be4b-e418-4176-8898-b53b00585604)
![image](https://github.com/user-attachments/assets/4b1234b3-427a-487f-a907-44cdf699f560)

But... it has no admin menu

![image](https://github.com/user-attachments/assets/f29cb020-88e5-42b8-8414-8b27531ddba9)

**Relevant Versions**
Nexus version: `3.75.1-01`
Plugin version: `3.2.0`

**Comment by tumbl3w33d** [comment]
While that might work (I would need to test it myself) the idea is to create an 'external role mapping' which connects a group/role that the user assumed in the identity provider with a Nexus equivalent. E.g., my operators have the administrator group in the IdP and I map it to a role in Nexus which includes nx-admin. Have you passed along group information via header? It is [expected to be a comma separated list](https://github.com/tumbl3w33d/nexus-oauth2-proxy-plugin/blob/fc1686933b1077e8a4b23817beb11b949a65de8a/src/main/java/com/github/tumbl3w33d/OAuth2ProxyRealm.java#L217).

**Comment by eirisdg** [comment]
> While that might work (I would need to test it myself) the idea is to create an 'external role mapping' which connects a group/role that the user assumed in the identity provider with a Nexus equivalent. E.g., my operators have the administrator group in the IdP and I map it to a role in Nexus which includes nx-admin. Have you passed along group information via header? It is [expected to be a comma separated list](https://github.com/tumbl3w33d/nexus-oauth2-proxy-plugin/blob/fc1686933b1077e8a4b23817beb11b949a65de8a/src/main/java/com/github/tumbl3w33d/OAuth2ProxyRealm.java#L217).
> 
> Make sure you created a new Nexus session after you directly assigned that original Nexus role. Might fix the issue. You can currently only logout by deleting cookies.

You are right. If you map the groups, the role is applied successful.
![image](https://github.com/user-attachments/assets/0fc65b71-7ce3-4634-a466-38ec3805864c)

BTW, the group mapping is optional parameter, so it should work without group.

**Comment by tumbl3w33d** [comment]
> BTW, the group mapping is optional parameter, so it should work without group.

I can confirm, this functionality was lost at some point. I guess we lost it when moving from Nexus' local user db to own tables for the user objects. Before, it was a side-effect that roles from the default realm kept being applied and all we needed to make sure is to not remove them in the sync process. Now it is the other way around: During the group sync, we would need to look for role assignments from the default user management.

Since the (idp) role mapping itself works, I guess this is not urgent and most users probably prefer that to manual local assignments anyway.

**Comment by eirisdg** [comment]
> I can confirm, this functionality was lost at some point. I guess we lost it when moving from Nexus' local user db to own tables for the user objects. Before, it was a side-effect that roles from the default realm kept being applied and all we needed to make sure is to not remove them in the sync process. Now it is the other way around: During the group sync, we would need to look for role assignments from the default user management.
> 
> Since the (idp) role mapping itself works, I guess this is not urgent and most users probably prefer that to manual local assignments anyway.
> 
> > BTW, the group mapping is optional parameter, so it should work without group.
> 
> You mean it is or should be optional to pass group information in from the idp? I must admit, I have only paid attention to the case that it is available and as long as this bug persists it is kind of required to make the setup usable.

Of course, I also think that all the attention should be focused on making the plugin as stable as possible for the use cases it's designed for. However, it might be interesting to correct the documentation and indicate that group mapping is mandatory and not optional.

I understand that efforts are directed where most people need them, but this change would avoid hours of testing.
